### ISSUE TITLE
Script to extract common data from Study doc and alert curators to new data that needs checking

### ISSUE THREAD

**Issue by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#1049]
### Inspiration

The new submission interface allows researchers (if they wish) to add compounds, tissues, treatments, LC Protocols, tracers, and infusates.  Some or all of these require manual curation by the curators who are loading the data.  These datums should be extracted from a submission and a consolidated doc should be updated.  Curators should be alerted to this new data and scrutinize is.  They should reach out to the researcher who submitted them if there are any problems, to coordinate resolutions.

### Description

Data that is common to and shared across all studies should be monitored for changes and manually scrutinized by curators.  We should create a script that:

1. Outputs the original xlsx study doc with common data sheets removed (e.g. the Compounds sheet)
2. Output a "consolidated_data.xlsx" doc with tabs for only the common data with warnings attached to new/changed data
3. Issues warnings for changes (new or modified) so the curator can scrutinize it (and those warnings applied as comments to affected rows)

Things to note:
- Deletion of rows will not delete data in the database, nor will there be an error.
- Modifications to rows would manifest as ConflictingValueErrors, if a load attempt was made in dry-run mode
- Checks for new data could either be a dry-run and an evaluation of the "created" stat or a `Model.objects.get()` call with all available fields supplied

An accompanying doc describing the load process from a curator perspective should accompany this effort.

### Alternatives

None

### Dependencies

This issue cannot be started until the completion of the following issue(s)/
pull request(s):

- #issue_number_1
- #pull_request_1

### Comment

None

-----

### Assumptions

1. List of assumptions that the code will not explicitly address/check
5. E.g. We will assume input is correct (explaining why there is no validation)

### Limitations

1. A list of things this work will specifically not do
2. E.g. This feature will only handle the most frequent use case X

### Affected Components

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

**Comment by mneinast** [comment]
Here's a relevant vignette from testing phase 1 of submission process that might highlight a way for users to update compound synonyms:

I tried uploading data containing compound names that do not exist as synonyms in tracebase (but the primary compound does exist).  The validate page caught this.  I then checked the Compounds tab in the study doc and found these problematic compounds:  they were missing an HMDB ID and the synonyms field was empty:
![image](https://github.com/user-attachments/assets/b1ba4c8f-dd0e-4536-a694-a154949b185d)

I decided to modify the Columns page in my Study Doc:  for each of these new synonyms, I added the correct HMDB ID:
![image](https://github.com/user-attachments/assets/5c026fbc-af18-47d4-898a-ca4eb01bec5d)

From the user's perspective this is an easy way to identify the compound unambiguously.  But I can see that this would create problems for the database (unless something is created that specifically handles this).  The validator found these and threw an error:

>  PASSED: Samples Check
PASSED: Peak Annotation Files Check
PASSED: Tissues Check
PASSED: Treatments Check
PASSED: Compounds Check
FAILED: col013b_study doc_240812.xlsx

    ConflictingValueErrors
    Conflicting values encountered during loading:
    	During the processing of sheet [Compounds] in col013b_study doc_240812.xlsx...
    	Creation of the following Compound record(s) encountered conflicts:
    		File record:     {'name': 'Glucose 6-phosphate', 'formula': 'C6H13O9P', 'hmdb_id': 'HMDB0001401'} (on row(s): 3)
    		Database record: {'name': 'glucose-6-phosphate', 'formula': 'C6H13O9P', 'hmdb_id': 'HMDB0001401'}
    			[name] values differ:
    			- database: [glucose-6-phosphate]
    			- file:     [Glucose 6-phosphate]
    		File record:     {'name': 'L-Lactic acid', 'formula': 'C3H6O3', 'hmdb_id': 'HMDB0000190'} (on row(s): 4)
    		Database record: {'name': 'lactate', 'formula': 'C3H6O3', 'hmdb_id': 'HMDB0000190'}
    			[name] values differ:
    			- database: [lactate]
    			- file:     [L-Lactic acid]

**Comment by hepcat72** [comment]
BTW, I think your comment should probably be in a different issue, but ignoring that for now...

@mneinast - So I think I can accurately guess what's going on here.  Here's what I expect happened.

- When you ran the start page to autofill, the compound synonym didn't exist.
- The compound existed, but it wasn't autofilled in the compounds sheet because the synonym search failed to find it
- So the autofill added a partial record based on the information it got from the peak annotation file (the synonym and the formula).  It created new rows for that data and could not fill in the HMDB ID nor the synonyms
- Then, when you went to validate, having added the HMDB ID, (which is required to be unique), you ended up with a conflicting value error.
- the solution is to add the synonyms from the Peak Annotations file to the existing compound records as synonyms instead of as separate compound records.
- The problem is that the autofill doesn't provide that compound record for you to edit.

I think that the solution here is to auto-populate all existing compounds and hope that the user will see that the autofilled row for what is essentially a compound synonym of another row and delete that row and add the synonym to the other record.

I'll create an issue for this.

**Comment by hepcat72** [comment]
#1153 created

**Comment by mneinast** [comment]
Yes i agree that's what happened.  Your proposed solution sounds good to me.
